### PR TITLE
Auth functionality

### PR DESCRIPTION
**Pull request by wiola-dreszler**

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -8,12 +8,14 @@ networks:
...
   frontend:
-    container_name: annotator_client
+    container_name: multi_annotator_client
     build:
       context: .
       dockerfile: ./client/Dockerfile
     ports:
-      - "8080:8080"
+      - 0.0.0.0:3000:3000
+      - 0.0.0.0:8080:8080
```
Comment (Tantael) [inline docker-compose.dev.yml]: Pick only one port that you actually need

File: client/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+WORKDIR /workspace/
+
+COPY ./client/package.json /workspace/package.json
+RUN npm install
+ENV NODE_PATH=/workspace/node_modules
+
+WORKDIR /workspace/client/
+
+EXPOSE 8080
```
Comment (Tantael) [inline client/Dockerfile]: Expose only port that you actually need

File: client/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+EXPOSE 3000
+CMD npm run start
```
Comment (Tantael) [inline client/Dockerfile]: add new line at end of file

File: client/src/Auth/Auth.tsx [hunk 4]
```diff
@@ -1,57 +1,88 @@
...
+                                account.
+                            </Typography>
+                            <Typography paragraph>
+                                If you have any questions please checkout
+                                the&nbsp;
+                                <Link href="https://github.com/jsbroks/Multi-annotator/wiki">
+                                    wiki&nbsp;
+                                </Link>
+                                before posting&nbsp;
+                                <Link href="https://github.com/jsbroks/Multi-annotator/issues">
```
Comment (Tantael) [inline client/src/Auth/Auth.tsx]: not existing link! change to https://github.com/OKEPL/multi-annotator/issues

File: docker-compose.dev.yml (hunk 1 above)
Comment (Tantael) [inline docker-compose.dev.yml]: Pick only one port that you actually need

**Review by Tantael — CHANGES_REQUESTED** [review]
Please resolve comments in review

File: client/src/Auth/auth.api.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    };
+    const response = await Api.request<RegisterResponse>(requestConfig);
+    if (response && response.data) {
+        return response.data;
+    }
+    return { error: response };
+};
+
+interface LoginResponse {
+    error?: string;
```
Comment (maciejdrozdowski) [inline client/src/Auth/auth.api.ts]: According to this interface, response from api might include only one error field, without information about success data fields. The same issue concern RegisterResponse...
I suggest to define ApiErrorInterface and make union type with success response interface.

File: client/src/Auth/auth.hooks.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    const [activeTab, setActiveTab] = useState<number>(0);
+    const [users, setUsers] = useState<number | undefined>(0);
+
+    const showLoginForm = users !== 0;
+
+    const fetchUsersInfo = async () => {
+        const usersInfo = await getServerInfo();
+        setUsers(usersInfo);
+    };
+    fetchUsersInfo();
```
Comment (maciejdrozdowski) [inline client/src/Auth/auth.hooks.tsx]: This method should be invoked inside useEffect hook ( if possible with "guard-param" or only once )

File: client/src/Auth/TabPanel.tsx [hunk 7]
```diff
@@ -12,55 +13,16 @@ interface Props {
...
+    const {
+        error,
+        userDetails: { fullName, username, password, confirmPassword },
+        isValidUsername,
+        isValidPassword,
+        isValidConfirmedPassword,
+        handleChange,
+        handleLogin,
+        handleRegister,
+    } = useTabPanel();
```
Comment (maciejdrozdowski) [inline client/src/Auth/TabPanel.tsx]: Hmmm... Perhaps it is a good idea to uplift this hook to Auth.tsx file and pass props like "loginMode","ButtonText" and "SubmitCb" instead passing "activeTab" and "index" and then decide internally which action should be performed ?

**Review by maciejdrozdowski — CHANGES_REQUESTED** [review]

File: client/src/Auth/auth.api.ts [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    git: {
+        tag: string;
+    };
+    total_users: number;
+    name: string;
+}
+
+export const getServerInfo = async () => {
+    const response = await Api.get<ServerInfoApiResponse>(serverInfoURL);
+    const totalUsers = response.data.total_users;
```
Comment (wojciech-bilicki) [inline client/src/Auth/auth.api.ts]: or just unpack the response?

File: client/src/Auth/auth.hooks.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+    const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
+        setActiveTab(newValue);
+    };
```
Comment (wojciech-bilicki) [inline client/src/Auth/auth.hooks.tsx:26]: new line missing

File: client/src/Auth/tabPanel.hooks.tsx [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+                password,
+            });
+
+            if (response && response.error) {
+                setError(response.error);
+            } else {
+                navigation.navigate('/');
+            }
+        }
+    };
```
Comment (wojciech-bilicki) [inline client/src/Auth/tabPanel.hooks.tsx]: new line missing

File: client/src/common/api.ts [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+                }
+            },
+        );
+    }
+
+    static get<T>(url: string, config?: AxiosRequestConfig) {
+        return this.axiosInstance.get<T>(url, config).catch(err => {
+            return err;
+        });
+    }
```
Comment (wojciech-bilicki) [inline client/src/common/api.ts:33]: new line

**Review by wojciech-bilicki — CHANGES_REQUESTED** [review]

**Review by wojciech-bilicki — DISMISSED** [review]

**Review by maciejdrozdowski — APPROVED** [review]

**Review by Tantael — APPROVED** [review]

**Pull request merged**